src/state_machine/auditor.zig

```diff
@@ -44,14 +44,20 @@ const PendingTransfer = struct {
 };
 
 const PendingExpiry = struct {
-    transfer: u128,
-    timestamp: u64,
+    transfer_id: u128,
+    transfer_timestamp: u64,
+    expires_at: u64,
 };
 
 const PendingExpiryQueue = PriorityQueue(PendingExpiry, void, struct {
-    /// Order by ascending timestamp.
+    /// Order by ascending expiration date and then by transfer's timestamp.
     fn compare(_: void, a: PendingExpiry, b: PendingExpiry) std.math.Order {
-        return std.math.order(a.timestamp, b.timestamp);
+        const order = switch (std.math.order(a.expires_at, b.expires_at)) {
+            .eq => std.math.order(a.transfer_timestamp, b.transfer_timestamp),
+            else => |order| order,
+        };
```

Comment:
Is there some Zig idiom that does essentially https://doc.rust-lang.org/std/cmp/enum.Ordering.html#method.then_with?

https://github.com/tigerbeetle/tigerbeetle/blob/main/src/scripts/cfo.zig#L555-L563 looks to ad-hoc to my taste to promote that to stdx...

Reply:
Not that I know of!